app/scripts/directives/rolefield.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+angular.module('bulbsCmsApp')
+  .directive('roleField', function ($http, routes, $, Raven) {
+    return {
+      templateUrl: routes.PARTIALS_URL + 'rolefield.html',
+      restrict: 'E',
+      replace: true,
+      scope: {
+        model: '='
+      },
+
+      link: function (scope, element, attrs) {
+        var resourceUrl = '/cms/api/v1/contributions/role/';
+
+        scope.roleValue = null;
+        scope.roleOptions = [];
+
+        scope.$watch('model.role', function () {
+          for (var i = 0; i < scope.roleOptions.length; i++) {
+            if (scope.roleOptions[i].id === Number(scope.roleValue)) {
+              scope.model.role = scope.roleOptions[i];
+            }
+          }
+          scope.roleValue = scope.model.role.id;
+        });
+
+        $http({
```

Comment:
All communication with the backend should occur through restmod. If possible, move this call into a restmod model.